libp2p/network/swarm.py

```diff
@@ -90,14 +91,49 @@ def set_stream_handler(self, stream_handler: StreamHandlerFn) -> None:
         except PeerStoreError:
             raise SwarmException(f"No known addresses to peer {peer_id}")
 
-        if not addrs:
+        if len(addrs) == 0:
             raise SwarmException(f"No known addresses to peer {peer_id}")
```

Comment:
It's actually still possible for IPeerStore to return an empty set, though this would not alter behavior much, iteration will not start, and it'll cascade direction to an exception. Though I would have to change wording for that to make sense.